test_rclcpp/test/test_timeout_subscriber.cpp

```diff
@@ -74,7 +77,7 @@ TEST_F(CLASSNAME(test_timeout_subscriber, RMW_IMPLEMENTATION), timeout_subscribe
     // ensure that the blocking spin does return after the specified timeout
     auto blocking_timeout = std::chrono::milliseconds(100);
     auto blocking_start = std::chrono::steady_clock::now();
-    executor.spin_node_once(node, blocking_timeout);
+    executor.spin_once(blocking_timeout);
```

Comment:
For this test, I think you'd have to:

- time a spin once
- if that spin once called the callback, check how long
- if not, loop

Basically spin one thing at a time, recording the time it took, and then check to see if it was the spin once we were looking for.

Reply:
mmm, I don't think this was testing that.
The callback is only throwing an exception https://github.com/ros2/system_tests/blob/11d32755043faf733ce4e0f6993d46b69ad9bcf0/test_rclcpp/test/test_timeout_subscriber.cpp#L33

IIUC what this is trying to test is that the timeout argument of `spin_once` is being used correctly.
But for actually checking that you have to make sure that no executable will be ready, which is pretty hard.

Honestly, I don't know why https://github.com/ros2/rclcpp/pull/1532 made this fail, seems to be random.

---

What I could do is spinning without adding any node to the executor ... I will try that.

Reply:
> What I could do is spinning without adding any node to the executor ... I will try that.

That also works, but really IDK if that was the intention of this test.

Reply:
To be honest, I think the original goal of the test is a bit misguided.

Maybe we could create a callback group for the subscription, add that the executor, then do the test as normal, therefore the things in the node will not affect it?

Reply:
That makes sense to me, see https://github.com/ros2/system_tests/pull/465/commits/f71c5524be5d034991b413dc5e4967badc878cef.